components/snackbar/Snackbar.js

```diff
@@ -39,15 +40,31 @@ const factory = (Overlay, Button) => {
     }
 
     componentWillReceiveProps (nextProps) {
+      const listenerTransition = () => {
+        this.setFocus();
+        this.refs.snackbar.removeEventListener('transitionend', listenerTransition);
+      };
+      if (nextProps.active) {
+        this.refs.snackbar.addEventListener('transitionend', listenerTransition);
```

Comment:
Isn't this adding tons event listeners? I think it should be checked if the listener is already added. May you look into it?